# QTM 385

***

## Homework 1

Student: Haozhou Huang[2343728]

***

### FAQ

1. What are the criteria for full marks?
 
> Two. Half of your grade is based on coding, and the other half is based on getting the results right.

2. How to write a clean code?

> Check out the PEP 8 Manual [here](https://www.python.org/dev/peps/pep-0008/). But in a nutshell:

> - Indents should be four spaces (not tab)
> - Limit all lines to a maximum of 79 characters (long blocks of text is limited to 72 characters)
> - Use a space after a comma
> - Use a space before and after arithmetic operators

3. How to get the question right?

> Answer to what has been asked. Not more, not less. Check twice that you did right and answered all the parts of the question. Like a good carpenter, measure twice, but cut only once: check your answers before submitting them.

4. How do I submit my homework?

> Two methods:

> - Upload your Jupyter Notebook to your Github page. Put the Github link on Canvas, and I'll check (recommended).

> - Upload your Jupyter Notebook on Canvas (not recommended).

5. What to do if I have other questions?

> Sign up for office hours using [this link.](https://calendly.com/umberto-mignozzetti/office-hours)

**Good luck!**


***

### Question 1

Run the following code:

```
0.1 + 0.1 + 0.1
```

Is it equal to `0.3`? If not, why?

In [5]:
0.1 + 0.1 + 0.1

0.30000000000000004

It is not equal to 0.3. Because the computer compute in decimal but in binary. The binary expression of the decimal number 0.1 is 0.000110011001100..., which has an infinite repeating fraction part. Computer stores this value as a double precision floating point number and approximate it to 0.00011001100110011001100110011001100110011001100110011001 based on IEEE 754 standard. By floating-point arithmetic, the computer adds them up and get the result 0.30000000000000004.


***

### Question 2

How many digits has $3^{1,000,000}$? (Hint: the function `str` transforms a number into a string.)

In [11]:
a=str(3**1000000)
len(a)

477122

There are total **477122** digits.


***

### Question 3

Create a tuple with three foods that you love to eat. Save it in a variable called `preferred_food`.

In [18]:
preferred_food = ('Fried Chicken','Hot Pot','Barbecue')
print(preferred_food)
type(preferred_food)

('Fried Chicken', 'Hot Pot', 'Barbecue')


tuple


***

### Question 4

The module random has a handy function for selecting elements in a tuple: `choice`. Import the module `random`, and name it `rd`.

In [20]:
import random as rd


***

### Question 5

Using the function choice, select a food to eat today. (Hint: when we used `numpy`, I named it `np`. When I needed a function in it, I did `np.function_I_needed`...)

In [23]:
rd.choice(preferred_food)

'Fried Chicken'

**That's all, folks!**